xonsh/jobs.py

```diff
@@ -6,13 +6,26 @@
 import signal
 import builtins
 from subprocess import TimeoutExpired, check_output
-from io import BytesIO
 from collections import deque
 
-from xonsh.tools import ON_WINDOWS
+from xonsh.platform import ON_DARWIN, ON_WINDOWS
 
 tasks = deque()
 
+
+if ON_DARWIN:
+    def _send_signal(job, signal):
+        # On OS X, os.killpg() may cause PermissionError when there are
+        # any zombie processes in the process group.
+        # See github issue #1012 for details
+        for pid in job['pids']:
+            os.kill(pid, signal)
+
+else:
+    def _send_signal(job, signal):
```

Comment:
When I wrote the code, I've already noticed it (defining it in Windows branch), but same thoughts as @adqm - 1) this is cleaner 2) Windows logics would never call it (this is Unix only thing).

We can change else to `elif not Windows`, which is make less readable but accurate.

Thoughts on change or not?